satellite/marketing/offers.go

```diff
@@ -0,0 +1,93 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information
+
+package marketing
+
+import (
+	"context"
+	"time"
+
+	"github.com/zeebo/errs"
+)
+
+// OffersErr creates offer error class
+var OffersErr = errs.Class("offers error")
+
+// Offers holds information about offer
+type Offers interface {
+	ListAll(ctx context.Context) ([]Offer, error)
+	GetCurrent(ctx context.Context, offerStatus OfferStatus, offerType OfferType) (*Offer, error)
+	Create(ctx context.Context, offer *NewOffer) (*Offer, error)
+	Update(ctx context.Context, offer *UpdateOffer) error
+}
+
+// NewOffer holds information that's needed for creating a new offer
+type NewOffer struct {
+	Name        string
+	Description string
+
+	AwardCreditInCents   int
+	InviteeCreditInCents int
+
+	RedeemableCap int
+
+	AwardCreditDurationDays   int
+	InviteeCreditDurationDays int
+
+	ExpiresAt time.Time
+
+	Status OfferStatus
+	Type   OfferType
+}
+
+// UpdateOffer holds fields needed for update an offer
+type UpdateOffer struct {
+	ID          int
+	Status      OfferStatus
+	NumRedeemed int
+	ExpiresAt   time.Time
+}
+
+// OfferType indicates the type of an offer
+type OfferType int
+
+const (
+	// FreeCredit is a type of offers used for Free Credit Program
+	FreeCredit OfferType = 0
+	// Referral is a type of offers used for Referral Program
+	Referral OfferType = 1
```

Comment:
You can use `iota` for enumeration, example https://github.com/storj/storj/blob/69d8b9f8286af2c7a4f32b57c036d5d84888e8e3/pkg/storj/encryption.go#L68